scripts/releaseNotes.mjs

```diff
@@ -105,6 +107,52 @@ const releaseNotes = (
 );
 console.error('📦 release notes JSON generated');
 
+const validateReleaseNotesAgents = (json) => {
+  // this set excludes 'sdk', 'node' and '.net' from the one above
+  const JSON_AGENTS = new Set([
+    'android',
+    'browser',
+    'dotnet',
+    'go',
+    'infrastructure',
+    'ios',
+    'java',
+    'nodejs',
+    'php',
+    'python',
+    'ruby',
+  ]);
+
+  JSON_AGENTS.forEach((agent) => {
+    const agentsCount = json.filter((note) => note.agent === agent).length;
+    if (agentsCount < 1) {
+      console.error(`😵 No release notes found for ${agent}`);
+      process.exitCode(1);
+    }
+    console.error(`🕵️ Found ${agentsCount} release notes for ${agent}`);
+  });
+
+  const requiredData = [
+    'agent',
+    'date',
+    'version',
+    'description',
+    'slug',
+    'eolDate',
+  ];
+
+  json.forEach((note) =>
+    requiredData.forEach((key) => {
+      if (!note[key]) {
+        console.error(note);
+        console.error(`😵 Missing ${key} data`);
+        process.exitCode(1);
+      }
+    })
+  );
+  console.error(`✨ Release notes JSON validated`);
```

Comment:
i think we should put a conditional here with a different message if validation fails. rn i think this kinda of reads like the notes are valid whether they pass or not

Reply:
Do you mean for eng readability? or for the Actions console? This line of code actually doesn't run if it [fails](https://github.com/newrelic/docs-website/actions/runs/6228948584/job/16906611920).

Reply:
updated! I actually like [this](https://github.com/newrelic/docs-website/actions/runs/6241138043/job/16942723937) better